packages/core/src/sfpcommands/source/DeploySourceToOrgImpl.ts

```diff
@@ -44,169 +43,70 @@ export default class DeploySourceToOrgImpl implements DeploymentExecutor {
-                        `sfdx force:mdapi:deploy:report --json -i ${deploy_id} -u ${this.target_org}`,
-                        {
-                            cwd: this.project_directory,
-                            stdio: ['pipe', 'pipe', 'ignore'],
-                            ...defaultProcessOptions(),
-                        }
-                    );
-                } catch (err) {
-                    if (this.deployment_options['checkonly'])
-                        SFPLogger.log(`Validation Failed`, LoggerLevel.ERROR, this.packageLogger);
-                    else SFPLogger.log(`Deployment Failed`, LoggerLevel.ERROR, this.packageLogger);
-                    break;
-                }
-
-                let resultAsJSON = JSON.parse(result);
-                if (resultAsJSON['status'] == 1) {
-                    SFPLogger.log('Deployment Failed', LoggerLevel.ERROR, this.packageLogger);
-                    commandExecStatus = false;
-                    break;
-                } else if (
-                    resultAsJSON['result']['status'] == 'InProgress' ||
-                    resultAsJSON['result']['status'] == 'Pending'
-                ) {
-                    SFPLogger.log(
-                        `${COLOR_TRACE(
-                            `Processing ${resultAsJSON.result.numberComponentsDeployed} out of ${resultAsJSON.result.numberComponentsTotal}`
-                        )}`,
-                        LoggerLevel.INFO,
-                        this.packageLogger
-                    );
-                } else if (resultAsJSON['result']['status'] == 'Succeeded') {
-                    SFPLogger.log(COLOR_SUCCESS('Deployment Succeeded'), LoggerLevel.INFO, this.packageLogger);
-                    commandExecStatus = true;
-                    break;
-                }
-
-                await delay(30000);
-            }
-
-            deploySourceResult.message = await this.getFinalDeploymentStatus(deploy_id);
-            deploySourceResult.result = commandExecStatus;
-            deploySourceResult.deploy_id = deploy_id;
             return deploySourceResult;
         }
     }
 
-    private async getFinalDeploymentStatus(deploy_id: string): Promise<string> {
+    private async displayErrors(result: DeployResult): Promise<string> {
         SFPLogger.log(`Gathering Final Deployment Status`, null, this.packageLogger);
-        let reportAsJSON = '';
-        let deploymentReports = `.sfpowerscripts/mdapiDeployReports`;
-        fs.mkdirpSync(deploymentReports);
 
-        try {
-            let child = child_process.exec(
-                `sfdx force:mdapi:deploy:report --json -i ${deploy_id} -u ${this.target_org} -w 30`,
-                {
-                    cwd: this.project_directory,
-                    ...defaultProcessOptions(),
-                }
+        if (result.response.numberComponentErrors > 0) {
+            DeployErrorDisplayer.printMetadataFailedToDeploy(
```

Comment:
Can change parameter on `printMetadataFailedToDeploy()` to be of type DeployMessage | DeployMessage[]